resources/qml/PrintMonitor.qml

```diff
@@ -20,49 +20,462 @@ Column
+                    {
+                        // Stacklevels
+                        // 0: user  -> unsaved change
+                        // 1: quality changes  -> saved change
+                        // 2: quality
+                        // 3: material  -> user changed material in materialspage
+                        // 4: variant
+                        // 5: machine_changes
+                        // 6: machine
+                        if ((bedTemperature.resolve != "None" && bedTemperature.resolve) && (bedTemperature.stackLevels[0] != 0) && (bedTemperature.stackLevels[0] != 1))
+                        {
+                            // We have a resolve function. Indicates that the setting is not settable per extruder and that
+                            // we have to choose between the resolved value (default) and the global value
+                            // (if user has explicitly set this).
+                            return bedTemperature.resolve;
+                        }
+                        else
+                        {
+                            return bedTemperature.properties.value;
+                        }
+                    }
+                    when: !preheatTemperatureInput.activeFocus
+                }
+            }
+        }
+
+        UM.RecolorImage
+        {
+            id: preheatCountdownIcon
+            width: UM.Theme.getSize("save_button_specs_icons").width
+            height: UM.Theme.getSize("save_button_specs_icons").height
+            sourceSize.width: width
+            sourceSize.height: height
+            color: UM.Theme.getColor("text")
+            visible: preheatCountdown.visible
+            source: UM.Theme.getIcon("print_time")
+            anchors.right: preheatCountdown.left
+            anchors.rightMargin: UM.Theme.getSize("default_margin").width / 2
+            anchors.verticalCenter: preheatCountdown.verticalCenter
+        }
+
+        Timer
+        {
+            id: preheatCountdownTimer
+            interval: 100 //Update every 100ms. You want to update every 1s, but then you have one timer for the updating running out of sync with the actual date timer and you might skip seconds.
+            running: false
+            repeat: true
+            onTriggered: update()
+            property var endTime: new Date() //Set initial endTime to be the current date, so that the endTime has initially already passed and the timer text becomes invisible if you were to update.
+            function update()
+            {
+                var now = new Date();
+                if (now.getTime() < endTime.getTime())
+                {
+                    var remaining = endTime - now; //This is in milliseconds.
+                    var minutes = Math.floor(remaining / 60 / 1000);
+                    var seconds = Math.floor((remaining / 1000) % 60);
+                    preheatCountdown.text = minutes + ":" + (seconds < 10 ? "0" + seconds : seconds);
```

Comment:
`Qt.formatTime` requires a `QTime` object or `QDateTime` object. I just have an integer in milliseconds. There is no "time interval" object in QT it seems (though [this stupid Wikipedia page](https://en.wikipedia.org/wiki/Qt_interval) ruins a lot of the searching I did on the internet). `QTime` wraps around at midnight. `QDateTime` can be used but constructing it requires constructing first a `QDateTime` object representing midnight of January 1st of the year 0, then adding a number of milliseconds to it, and that just seems like overkill.

This can all be moved into Python, luckily.